projects/vdk-core/src/vdk/internal/builtin_plugins/connection/impl/router.py

```diff
@@ -75,25 +75,24 @@ def open_connection(self, dbtype: str) -> ManagedConnectionBase:
         :return: the new connection if succesfull or throws an expception
         """
         dbtype = dbtype.lower() if dbtype else None
+        conn = None
         if dbtype in self._connections:
             conn = self._connections[dbtype]
-            if conn._is_connected():
-                return conn
-            else:
-                conn._connect()
-                return conn
         self._log.debug(f"Connection to {dbtype} is missing. Will try to connect")
         if dbtype in self._connection_builders:
-            return self.__create_connection(dbtype)
-        errors.log_and_throw(
-            to_be_fixed_by=errors.ResolvableBy.CONFIG_ERROR,
-            log=self._log,
-            what_happened=f"Provided configuration variable for {DB_DEFAULT_TYPE} has invalid value.",
-            why_it_happened=f"VDK was run with {DB_DEFAULT_TYPE}={dbtype}, however {dbtype} is invalid value for this variable.",
-            consequences=errors.MSG_CONSEQUENCE_DELEGATING_TO_CALLER__LIKELY_EXECUTION_FAILURE,
-            countermeasures=f"Provide either valid value for {DB_DEFAULT_TYPE} or install database plugin that supports this type. "
-            f"Currently possible values are {list(self._connection_builders.keys())}",
-        )
+            conn = self.__create_connection(dbtype)
+        else:
+            errors.log_and_throw(
+                to_be_fixed_by=errors.ResolvableBy.CONFIG_ERROR,
+                log=self._log,
+                what_happened=f"Provided configuration variable for {DB_DEFAULT_TYPE} has invalid value.",
+                why_it_happened=f"VDK was run with {DB_DEFAULT_TYPE}={dbtype}, however {dbtype} is invalid value for this variable.",
+                consequences=errors.MSG_CONSEQUENCE_DELEGATING_TO_CALLER__LIKELY_EXECUTION_FAILURE,
+                countermeasures=f"Provide either valid value for {DB_DEFAULT_TYPE} or install database plugin that supports this type. "
+                f"Currently possible values are {list(self._connection_builders.keys())}",
+            )
+        conn.connect()
```

Comment:
`conn._connect()` was used priorly,
now by using `connect()` instead then wouldn't you, again, skip the reconnecting part due to:
```
def connect(self) -> PEP249Connection:
        """
        :return: PEP249 Connection object (managed)
        """
        if not self._is_db_con_open:
            db_con = self._connect()
            self._log.debug(f"Established {str(db_con)}")
            self._is_db_con_open = True
            self._db_con = db_con

        return self
```

How is this "reconnect" to return a verified to be opened connection, tested?

Reply:
The new functional test covers this.

Reply:
did checkout the branch to see the reconnect made, looks good to me